File: 2576-minimum-penalty-for-a-shop/minimum-penalty-for-a-shop.java
class Solution {
    public int bestClosingTime(String customers) {
        int c=0,max=0,minp=0;
        for(int i=0;i<customers.length();i++){
            char ch=customers.charAt(i);
            if(ch=='Y') c++;
            else c--;
            if(max<c){
                minp=i+1;
                max=c;
            }
        }
        return minp;
    }
}

File: 2576-minimum-penalty-for-a-shop/minimum-penalty-for-a-shop.js
/**
 * @param {string} customers
 * @return {number}
 */
var bestClosingTime = function(customers) {
    let c=0,max=0,minp=0;
    for(var i=0;i<customers.length;i++){
        if(customers[i]=='Y') c++;
        else c--;
        if(max<c){
            minp=i+1;
            max=c;
        }
    }
    return minp;
};